From david-b@pacbell.net Mon Nov 28 09:33:38 2005
From: Chris Humbert <mahadri-kernel@drigon.com>
Subject: USB: don't allocate dma pools for PIO HCDs
Date: Mon, 28 Nov 2005 09:29:23 -0800
Cc: David Brownell <david-b@pacbell.net>
Content-Disposition: inline
Message-Id: <200511280929.23666.david-b@pacbell.net>

USB: don't allocate dma pools for PIO HCDs

hcd_buffer_alloc() and hcd_buffer_free() have a similar dma_mask
check and revert to kmalloc()/kfree(), but hcd_buffer_create()
doesn't check dma_mask and allocates unused dma pools.

Signed-off-by: Chris Humbert <mahadri-kernel@drigon.com>
Acked-by: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/buffer.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/core/buffer.c
+++ gregkh-2.6/drivers/usb/core/buffer.c
@@ -55,6 +55,9 @@ int hcd_buffer_create (struct usb_hcd *h
 	char		name [16];
 	int 		i, size;
 
+	if (!hcd->self.controller->dma_mask)
+		return 0;
+
 	for (i = 0; i < HCD_BUFFER_POOLS; i++) { 
 		if (!(size = pool_max [i]))
 			continue;
